src/StrawberryShake/Client/src/Transport.WebSockets/DefaultSocketClientFactory.cs

```diff
@@ -0,0 +1,38 @@
+using System;
+using System.Linq;
+using Microsoft.Extensions.Options;
+
+namespace StrawberryShake.Transport.WebSockets
+{
+    public class DefaultSocketClientFactory
+        : ISocketClientFactory
+    {
+        private readonly IOptionsMonitor<SocketClientFactoryOptions> _optionsMonitor;
+
+        public DefaultSocketClientFactory(
+            IOptionsMonitor<SocketClientFactoryOptions> optionsMonitor)
+        {
+            _optionsMonitor = optionsMonitor;
+        }
+
+        public ISocketClient CreateClient(string name)
+        {
+            if (string.IsNullOrEmpty(name))
+            {
+                throw new ArgumentException(
+                    "The web socket name cannot be null or empty.",
+                    nameof(name));
+            }
+
+            SocketClientFactoryOptions options = _optionsMonitor.Get(name);
+            var client = new WebSocketClient(name, options.Protocols.ToArray());
```

Comment:
At the moment there is only a websocket factory. But theoretically the `ISocketClient` is abstract an can be anything
We could specify the specific factory in `SocketClientFactoryOptions` but as i am not sure if we even need this i just left it this way